extensions/amp-form/0.1/amp-form.js

```diff
@@ -792,7 +792,17 @@ export class AmpFormService {
    * @param  {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
    */
   constructor(ampdoc) {
-    this.installStyles_(ampdoc).then(() => this.installHandlers_(ampdoc));
+    /** @const @private {!Promise} */
+    this.whenFinished_ = this.installStyles_(ampdoc)
+        .then(() => this.installHandlers_(ampdoc));
+  }
+
+  /**
+   * Returns a promise that resolves when all form implementations (if any)
+   * have been upgraded.
+   */
+  whenFinished() {
```

Comment:
Not sure if `whenFinished` is a good name since this part is re-entrant. Maybe something like "whenInitialized"?